util/timer_test.cc

```diff
@@ -390,6 +390,45 @@ TEST_F(TimerTest, DestroyTimerWithRunningFunc) {
   control_thr.join();
 }
 
+TEST_F(TimerTest, ConcurrentStartShutdown) {
```

Comment:
Would be great to add a deterministic test by using `mock_env_.MockSleepForMicroseconds()` + `TEST_SYNC_POINT()` in timer.h, for example:
https://github.com/facebook/rocksdb/blob/9be75629b4cc5fa0786440f7c92e4f67ca3fdebe/util/timer_test.cc#L178-L183